### ISSUE TITLE
Graphite fails to render a graph because it divides something by zero

### ISSUE THREAD

**Issue by Al2Klimov** [Icinga/icingaweb2-module-graphite#79]
## Steps to Reproduce (for bugs)

Go to the graphs list for hosts or services and look out for headlines with neither any graphs nor "No graphs found" below them.

## See also

https://github.com/brutasse/graphite-api/issues/116

**Comment by Al2Klimov** [comment]
@bobapple @lippserd What's your opinion on how we should deal w/ this?

**Comment by Al2Klimov** [comment]
Anyway I'd stall this in favor of all the other stuff.

**Comment by Al2Klimov** [comment]
As discussed w/ @bobapple (offline):

We should close this as we can't reproduce it.

**Comment by Al2Klimov** [comment]
As investigated w/ @lippserd and @gethash: It still occurrs!

**Comment by Al2Klimov** [comment]
Saw this today again. Will build a Vagrantfile w/ everything needed to reproduce this for the Graphite devs.

**Comment by Al2Klimov** [comment]
Built [the Vagrant box](https://github.com/Al2Klimov/graphite-web-div0). (Finally!) Could not reproduce this again. I'll give up at this point as this is not critical at all – while @gethash was preparing his demo env, this happened only on the first few minutes. (IMO it's not "line mode: connected".) If (the Graphite Web devs didn't already fix that and) someone discovers this again, they will open an issue here.

CC @bobapple

**Comment by dnsmichi** [comment]
I'm able to reproduce this here with Graphite 1.1.1 and the following request URL.

```
2018-02-12,15:16:59.084 :: Exception encountered in <POST http://192.168.33.5:8003/render?from=1518444959&until=1518445019&width=350&height=200&hideLegend=&tz=Europe%2FParis&_salt=1518445019.000&vTitle=Percent&lineMode=connected&drawNullAsZero=false&graphType=line&majorGridLineColor=%230000003F&minorGridLineColor=%2300000000&_ext=whatever.svg&areaAlpha=0.5&areaMode=all&lineWidth=2&min=0&yUnitSystem=binary&target=alias%28color%28icinga2.icinga2.services.http_Icinga_Website.http.perfdata.size.value%2C%20%27%231a7dd7%27%29%2C%20%27Response%20size%20%28bytes%29%27%29>
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/graphite/webapp/graphite/render/views.py", line 144, in renderView
    response = renderViewGraph(graphOptions, requestOptions, data)
  File "/opt/graphite/webapp/graphite/render/views.py", line 164, in renderViewGraph
    image = doImageRender(requestOptions['graphClass'], graphOptions)
  File "/opt/graphite/webapp/graphite/render/views.py", line 576, in doImageRender
    img = graphClass(**graphOptions)
  File "/opt/graphite/webapp/graphite/render/glyph.py", line 574, in __init__
    self.drawGraph(**params)
  File "/opt/graphite/webapp/graphite/render/glyph.py", line 1076, in drawGraph
    self.consolidateDataPoints()
  File "/opt/graphite/webapp/graphite/render/glyph.py", line 1419, in consolidateDataPoints
    bestXStep = numberOfPixels / divisor
ZeroDivisionError: float division by zero
```

**Comment by Al2Klimov** [comment]
Is that behavior consistent? Does it still occur after your Icinga 2 kept feeding data into Graphite?

**Comment by dnsmichi** [comment]
It is a fresh Vagrant box which is now running for round about 30 minutes.

I did some research, https://github.com/graphite-project/graphite-web/issues/1245 puts a valid point:

```
So it then has a time range that is lower than the largest step. Then integer division happens and it crashes.
```

If I change the timerange from the default to 1h, graphs are properly rendered. 

Probably it is a wrong configuration of mine, using 1 minute as default time range. 

```
[graphite]
url = "http://192.168.33.5:8003"

[ui]
default_time_range = "1"
default_time_range_unit = "minutes"
disable_no_graphs_found = "0"
```

This is something I may choose from the web configuration form.

![screen shot 2018-02-12 at 15 46 03](https://user-images.githubusercontent.com/382049/36102202-e5b61544-100b-11e8-9f18-50c9d32d9d22.png)


The dropdown inside the service detail view doesn't go lower than 5 minutes (which then works fine ootb).

![screen shot 2018-02-12 at 15 44 10](https://user-images.githubusercontent.com/382049/36102162-ca829928-100b-11e8-865b-8b942ec82698.png)

So I'd say I made a configuration issue, and the form shouldn't allow me to specify a default time range lower than 5 minutes :)
